### PR TITLE
Fix typos in temporary table docs

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Fix typos in temporary table docs

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/7402

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by lilin90 — APPROVED** [review]
LGTM

**Review by Liuxiaozhen12 — APPROVED** [review]

File: temporary-table.md [hunk 1]
```diff
@@ -180,7 +181,8 @@ TiDB 本地临时表与 MySQL 临时表有以下方面不兼容：
 
 > **注意：**
 >
-> 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他-tidb-功能的兼容性限制)。
+> - 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他-tidb-功能的兼容性限制)以及[与 MySQL 临时表的兼容性](#与-mysql-临时表的兼容性)。
```
Comment (TomShawn) [inline temporary-table.md:184]:
```suggestion
> - 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他-tidb-功能的兼容性限制)。
```

File: temporary-table.md [hunk 2]
```diff
@@ -43,7 +43,8 @@ TiDB 的临时表分为本地临时表和全局临时表：
 
 > **注意：**
 >
-> 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他-tidb-功能的兼容性限制)以及[与 MySQL 临时表的兼容性](#与-mysql-临时表的兼容性)。
+> - 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他-tidb-功能的兼容性限制)以及[与 MySQL 临时表的兼容性](#与-mysql-临时表的兼容性)。
+> - 如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级后也会被 TiDB 当成普通表处理。
```
Comment (en-jin19) [inline temporary-table.md:47]:
```suggestion
> - 如果在 v5.3.0 升级前创建了本地临时表，这些临时表实际为普通表，在升级后也会被 TiDB 当成普通表处理，导致数据错误。
```

Comment (en-jin19) [inline temporary-table.md:47]: 本地临时表会有全局临时表中的降级后会导致数据错误的情况吗？

File: temporary-table.md [hunk 3]
```diff
@@ -180,7 +181,8 @@ TiDB 本地临时表与 MySQL 临时表有以下方面不兼容：
 
 > **注意：**
 >
-> 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他-tidb-功能的兼容性限制)。
+> - 使用 TiDB 中的临时表前，注意临时表[与其他 TiDB 功能的兼容性限制](#与其他-tidb-功能的兼容性限制)。
+> - 在 v5.3.0 上创建的全局临时表在降级后会被当作普通表处理，导致数据错误。
```
Comment (en-jin19) [inline temporary-table.md:185]: 与本地临时表的描述保持了统一
```suggestion
> - 如果在 v5.3.0 上创建了全局临时表，这些临时表在降级后会被当作普通表处理，导致数据错误。
```

Comment (TomShawn) [reply temporary-table.md]: addressed, thanks~

**Pull request merged**